core/ArchiveProcessor/PluginsArchiver.php

```diff
@@ -114,25 +115,32 @@ public function callAggregateAllPlugins($visits, $visitsConverted)
 
                 $this->logAggregator->setQueryOriginHint($pluginName);
 
-                $timer = new Timer();
-                if ($this->isSingleSiteDayArchive) {
-                    Log::debug("PluginsArchiver::%s: Archiving day reports for plugin '%s'.", __FUNCTION__, $pluginName);
+                try {
+                    $timer = new Timer();
+                    if ($this->isSingleSiteDayArchive) {
+                        Log::debug("PluginsArchiver::%s: Archiving day reports for plugin '%s'.", __FUNCTION__, $pluginName);
 
-                    $archiver->aggregateDayReport();
-                } else {
-                    Log::debug("PluginsArchiver::%s: Archiving period reports for plugin '%s'.", __FUNCTION__, $pluginName);
+                        $archiver->aggregateDayReport();
+                    } else {
+                        Log::debug("PluginsArchiver::%s: Archiving period reports for plugin '%s'.", __FUNCTION__, $pluginName);
 
-                    $archiver->aggregateMultipleReports();
-                }
+                        $archiver->aggregateMultipleReports();
+                    }
+
+                    $this->logAggregator->setQueryOriginHint('');
 
-                $this->logAggregator->setQueryOriginHint('');
+                    Log::debug("PluginsArchiver::%s: %s while archiving %s reports for plugin '%s'.",
+                        __FUNCTION__,
+                        $timer->getMemoryLeak(),
+                        $this->params->getPeriod()->getLabel(),
+                        $pluginName
+                    );
+                } catch (Exception $e) {
+                    $Klass = get_class($e);
+                    $exception = new $Klass($e->getMessage() . " - caused by plugin $pluginName", $e->getCode(), $e);
```

Comment:
Not sure why I wrote `$Klass` :) Think we used to do it like this in another project when it contained classname... will rename